posts/TEMPLATE/index.md

```diff
@@ -1,9 +1,117 @@
+title: Quickly test whether ports can be reached on your linux machine.
+author: julian.markwort@cybertec.at
+tags: ["firewall", "linux", "netcat", "nc", "ports", "testing", "probe", "listening address"] # max. 10 tags; lowercase; dash-separated
+description: "How would you quickly check if a given port is reachable on your system or not, so you can go complain to the network admins? I'll show you my way." # max. 300 chars.
 ---
 
-This is just an example post, don't worry about it. We use this internally as a template for awesome, future posts.
+If you've ever tried to launch anything that opens a port on your linux machine and you'd like to know whether you've misconfigured something (e.g. the listen address), or wether you're running into firewall issues, you'll be grateful to find out that there's a very simple way to test for firewall issues.
+
+There's a tool called `netcat`; It's part of almost any linux server setup I've worked on and it can be launched by typing `nc` on your command line.
+There are two modes to netcat:
+- a calling mode
+- a listening mode
+
+We can use both of these, on two sides of our server to check if the firewall will let us pass or not.
+
+(insert thou shall not pass image)
+
+## Introduction to netcat
+
+On the machine where you'd like to run the server, which will be listen on a given port, say 54321, run this to open a listening netcat:
+```
+nc -vlk 54321
+```
+I'll break those arguments down for you:
+- v is for verbose: print more details about connections
+- l is for listening mode
+- k is for keep-open; This means that netcat will accept connections over and over again.
+- 54321 is for the listening port. by default, netcat listens on all IP-addresses it can find, but you can also limit it to a given IP-address, say your primary IP-address, so your service can be reached from your network, but not via localhost, for example.
+
+
+Vou can stop this netcat by sending the Interrupt Signal, which is usually achieved by pressing `Ctrl` and `c` at the same time in your terminal session.
+
+Now, on the second machine, we'll launch another netcat, this one is set to connect to the first machine, by IP-address (or even hostname) and by port:
+```
+nc -vz 192.168.178.24 54321
+```
+Here, the arguments have the following meaning:
+- v is still for verbose...
+- z is for Zero-I/O mode, meaning that netcat will initiate a connection and report success or failure, but won't transmit any data
+- 192.168.178.24 is my first machine's local IP-address
+- 54321 is the port that we'd like to connect to.
+
+> Please be aware that it's never possible to bind two applications to the same port. So as long as you have a netcat running that listens on port 54321, trying to launch another application that should listen on 54321 will fail. If in doubt, run `sudo killall nc` to kill all netcats that may still be running in the background (If you launched the netcats detached, by adding a postfix ` &` to your `nc` call).
```

Comment:
```suggestion
> Please be aware that it's impossible to bind two applications to the same port. So as long as you have a netcat instance running that listens on port 54321, trying to launch another application that should listen on 54321 will fail. If in doubt, run `sudo killall nc` to kill all netcat instances that might still be running in the background (in case netcat was launched detached, by adding a postfix ` &` to your `nc` call).
```

Reply:
wie hier...
https://github.com/cybertec-postgresql/today-i-learned-content/pull/43/files/454ec75a9954afcc1d19e32174d792d67a25c093#r329319511

Reply:
Edited to only correct "never possible"